serf/merge_delegate.go

```diff
@@ -45,5 +56,31 @@ func (m *mergeDelegate) nodeToMember(n *memberlist.Node) *Member {
 		DelegateMin: n.DMin,
 		DelegateMax: n.DMax,
 		DelegateCur: n.DCur,
+	}, nil
+}
+
+// validateMemberInfo checks that the data we are sending is valid
+func (m *mergeDelegate) validiateMemberInfo(n *memberlist.Node) error {
+	var InvalidNameRe = regexp.MustCompile(`[^A-Za-z0-9\\-]+`)
+	var InvalidIPv4Re = regexp.MustCompile(`([0-9]{1,3}\.){3}[0-9]{1,3}:[0-9]+`)
+	var InvalidIPv6Re = regexp.MustCompile(`([0-9a-f]{1,4}:){7}[0-9a-f]{1,4}`)
```

Comment:
I think instead of using regexes for validating the address/port we could do something like this: 
```
host, port, err := net.SplitHostPort(n.Addr)
if err != nil {
   return err
}

ip := net.ParseIP(host)
if ip == nil || (ip.To4() == nil && ip.To16() == nil) {
    fmt.Printf("%v is not a valid IPv4 or 1Pv6 address\n", ip)
    return false
}

p, err := strconv.Atoi(port)
if err != nil {
    return err
}
if p < 0 || p > 65535 {
    return fmt.Errorf("port must be a valid number from 0-65535")
} 
```

This way we can use the go stdlib stuff to validate the IP address format.